internal/users/manager.go

```diff
@@ -17,6 +17,10 @@ import (
 	"github.com/ubuntu/decorate"
 )
 
+const (
+	maxPreAuthUsers = 1024
```

Comment:
This seems big enough, but I feel we can be even more generous, no?

Reply:
decreased to 4096 because of:

```
// MaxPreAuthUsers is the maximum number of pre-auth users that can be registered. If this limit is reached,
// RegisterPreAuthUser will return an error and disable login for new users via SSH until authd is restarted.
//
// This value must be significantly smaller (less than half) than the number of UIDs which can be generated (as
// defined by UID_MIN and UID_MAX in the config file), otherwise finding a unique UID by trial and error can take
// too long.
```

... and I feel like we should support a range of 100000. I would be fine with increasing it up to 32768 if you still prefer a higher number.